src/stepfunctions/steps/service.py

```diff
@@ -887,3 +892,59 @@ def __init__(self, state_id, **kwargs):
+
+
+class StepFunctionsStartExecutionStep(Task):
+
+    """
+    Creates a Task state that starts an execution of another state machine. See `Manage AWS Step Functions Executions as an Integrated Service <https://docs.aws.amazon.com/step-functions/latest/dg/connect-stepfunctions.html`_ for more details.
+    """
+
+    def __init__(self, state_id, wait_for_callback=False, wait_for_completion=True, json_ouput=False, **kwargs):
+        """
+        Args:
+            state_id (str): State name whose length **must be** less than or equal to 128 unicode characters. State names **must be** unique within the scope of the whole state machine.
+            wait_for_callback(bool, optional): Boolean value set to `True` if the Task state should wait for callback to resume the operation. (default: False)
+            timeout_seconds (int, optional): Positive integer specifying timeout for the state in seconds. If the state runs longer than the specified timeout, then the interpreter fails the state with a `States.Timeout` Error Name. (default: 60)
+            timeout_seconds_path (str, optional): Path specifying the state's timeout value in seconds from the state input. When resolved, the path must select a field whose value is a positive integer.
+            heartbeat_seconds (int, optional): Positive integer specifying heartbeat timeout for the state in seconds. This value should be lower than the one specified for `timeout_seconds`. If more time than the specified heartbeat elapses between heartbeats from the task, then the interpreter fails the state with a `States.Timeout` Error Name.
+            heartbeat_seconds_path (str, optional): Path specifying the state's heartbeat value in seconds from the state input. When resolved, the path must select a field whose value is a positive integer.
+            comment (str, optional): Human-readable comment or description. (default: None)
+            input_path (str, optional): Path applied to the state’s raw input to select some or all of it; that selection is used by the state. (default: '$')
+            parameters (dict, optional): The value of this field becomes the effective input for the state.
+            result_path (str, optional): Path specifying the raw input’s combination with or replacement by the state’s result. (default: '$')
+            output_path (str, optional): Path applied to the state’s output after the application of `result_path`, producing the effective output which serves as the raw input for the next state. (default: '$')
+            wait_for_completion (bool, optional): Boolean value set to `True` if the Task state should wait to complete before proceeding to the next step in the workflow. (default: True)
+            json_output(bool, optional): Boolean value set to `True` if the the Task should return a response in JSON format. If set to `False`, the Task should return a str response (default: False)
```

Comment:
>- default to sync:2

Agreed - let's default to sync:2 - I think we should be consistent with the CDK and this seems like it would be the most used resource between the two.

>- don't make json_output configurable

Since `startExecution.sync` is in our documentation, I believe it could be good to provide an option for our customers to use it. How about instead of being `json_output` flag, we provide a flag to opt in for a string response instead? Something like `string_response` or `string_output` that can be set in the kwargs. I'm curious to see what other reviewers think too and will go with the majority

Reply:
> Since startExecution.sync is in our documentation, I believe it could be good to provide an option for our customers to use it.

It will always be in the documentation as it was the only option available initially. As we refined our approach and offered a better alternative, `sync:2` was introduced. 

We shouldn't seek to hit arbitrarily high 1:1 parity with documentation. Rather, we should ask ourselves what use cases it unlocks, whether `sync:2` precludes the ability to do anything that `sync` offers. Do you have a use case in mind?

We have not exposed this integration to customers yet, so my recommendation is still to start as tight as possible and expand it in the future if we have a feature request / use case in mind. Nothing prevents us from exposing it in a future release, but supporting it at the outset based on assumptions is fragile and also a 1-way door as we'll have it out there forever (at least until v3).

thoughts?

Reply:
>We have not exposed this integration to customers yet, so my recommendation is still to start as tight as possible and expand it in the future if we have a feature request / use case in mind. Nothing prevents us from exposing it in a future release, but supporting it at the outset based on assumptions is fragile and also a 1-way door as we'll have it out there forever (at least until v3).

I definitely agree with this - we shouldn't introduce 1 way doors. I don't have as specific use case in mind where `:sync` would be preferred over `:sync:2`. If there is ask for it, we can always add support for it later and it won't be a breaking change

Reply:
> we shouldn't introduce 1 way doors.

That's not quite what I meant. I meant that when we go through a 1-way door, we should have a deep understanding of **why** we are doing it.